### DISCUSSION TITLE
How do you add padding between paragraphs and bulleted items (and a few other styling questions)?

### DISCUSSION THREAD

**Discussion by lancejpollard** [dolanmiu/docx#2589] [Q&A]
Hi there, I am making a markdown to docx converter using your library yay :) https://github.com/lancejpollard/md2docx.js/blob/make/index.ts . See the simple `doc` at the bottom for the styles I applied from your documentation.

I have some questions about how to style some stuff.

1. How do you add padding/spacing between paragraphs?
2. How do you add padding before/after a whole list (but not list items or nested lists)?
3. How do you neatly indent any combination of nested numbered/bulleted lists so each list item is 16px (or equivalent) from the left, and nested ones are 16px further from the parent?
4. How do you do links? My `ExternalLink` usage doesn't seem to be working.
5. How do you customize the bullet on bulleted lists to be any unicode symbol or maybe even just an asterisk?

Basically I want like 16px of spacing between paragraphs and lists in a document, and to customize the spacing underneath and above headers based on their font size.

My document looks like this so far, but I would like it to look nicer.

<img width="649" alt="Screenshot 2024-02-10 at 9 43 57 PM" src="https://github.com/dolanmiu/docx/assets/28718/bd6016b6-e500-4cba-af78-a76185c44501">

Thanks for the help!

_Cross-posted to StackOverflow: https://stackoverflow.com/questions/77975860/how-to-change-styling-of-paragraphs-and-list-items-in-docx-for-javascript_

**Comment by lancejpollard** [comment]
I got it working in the press-SPACE preview on Mac:

<img width="564" alt="Screenshot 2024-02-11 at 1 42 45 AM" src="https://github.com/dolanmiu/docx/assets/28718/e1dfe1de-a03f-4fde-98cf-d8cbb23ad77b">

But when I open it up in Pages on Mac:

<img width="652" alt="Screenshot 2024-02-11 at 1 42 53 AM" src="https://github.com/dolanmiu/docx/assets/28718/2a437bce-f54b-410a-8d11-9993d64f7e2d">

How to fix?

I'm using:


```js
{
  level: 0,
  format: LevelFormat.BULLET,
  text: '•',
  alignment: AlignmentType.LEFT,
  style: {
    paragraph: {
      indent: {
        start: convertInchesToTwip(INDENT * 0),
      },
    },
  },
}
```

And then on Google Docs, sigh:

<img width="701" alt="Screenshot 2024-02-11 at 1 45 10 AM" src="https://github.com/dolanmiu/docx/assets/28718/97194130-32a5-4563-86b0-59df532b9d6f">
